docker/generic/Dockerfile

```diff
@@ -0,0 +1,52 @@
+FROM alpine:3.7
+MAINTAINER Emmanuel Frecon <efrecon@gmail.com>
+
+LABEL Description="Eclipse Mosquitto MQTT Broker"
+
+ARG VERSION=1.4.15
+
+RUN set -ex \
+    # Add build dependencies, remove after build
+    && apk --no-cache add --virtual .build-deps \
+        build-base \
+        libressl-dev \
+        c-ares-dev \
+        util-linux-dev \
+        libwebsockets-dev \
+    # Add fetch dependencies, remove after build
+    && apk --no-cache add --virtual .fetch-deps \
+        git \
+        curl \
+    # Add run dependencies, keep after build
+    && apk --no-cache add --virtual .run-deps \
+        libressl \
+        c-ares \
+        util-linux \
+        libwebsockets \
+    # Checkout and compile, versions are branched with a "v" in front of the
+    # version number, but tagged at the Docker registry without the "v".
+    && git clone --depth 1 -b v${VERSION} https://github.com/eclipse/mosquitto.git /tmp/mosquitto \
+    && cd /tmp/mosquitto \
+    && curl -qsL https://git.alpinelinux.org/cgit/aports/plain/main/mosquitto/libressl.patch -o libressl.patch \
+    && git apply libressl.patch \
+    && make \
+        WITH_MEMORY_TRACKING=no \
+        WITH_WEBSOCKETS=yes \
+        WITH_SRV=yes \
+        WITH_TLS_PSK=no \
+        WITH_ADNS=no \
+        WITH_DOCS=no \
+        prefix=/usr \
+    && sed -i -e "s|(INSTALL) -s|(INSTALL)|g" -e 's|--strip-program=${CROSS_COMPILE}${STRIP}||' */Makefile */*/Makefile \
```

Comment:
what's the purpose of this line? Is this really indicating a makefile deficiency that should be addressed separately?

Reply:
Yes, at least, I thought it was, and I now see that it has been fixed by d90cd585dd8cc1f9cecd9082d71333e1e638df2d. I hadn't seen that before you pointed it out. Again, this behaviour is inherited from the way Alpine does it when [compiling](https://git.alpinelinux.org/cgit/aports/plain/main/mosquitto/APKBUILD?h=3.4-stable) mosquitto.

Reply:
Yes, copying things blindly from some random downstream distro is not necessarily a path to glory.